### DISCUSSION TITLE
What happens when message cycle time is None?

### DISCUSSION THREAD

**Discussion by superaga** [cantools/cantools#570] [Q&A] [answered]
Hi all!
The [Message](https://cantools.readthedocs.io/en/latest/#cantools.database.can.Message) class doesn't require that the _message cycle time_ property would be specified to instantiate the object, so I'm wondering how this would be treated from the CAN node.
Is this never sent or is this sent at the "default" cycle time (that depends by the device)?
If this property is necessary shouldn't this set as mandatory?

Many thanks!

**Comment by andlaus** [comment, marked as answer]
this means that the cycle time is not specified. This may be because the message is only send when a certain event happens, the message may be not for signal based communication (e.g., UDS diagnostics),  its schedule follows e.g., a Fibonacci sequence instead of  a fixed intervals, or the person who created the database just did not specify the cycle time.

> If this property is necessary shouldn't this set as mandatory?

it is not mandatory nor does it always make sense...

**Comment by superaga** [comment]
Hi @andlaus,

> This may be because the message is only send when a certain event happens

I didn't know this, it is very useful!
Sure, now makes sense.

Many thanks
